### ISSUE TITLE
Error in "injectable-should-be-provided" Rule for Factory Providers

### ISSUE THREAD

**Issue by MR-Mostafa** [darraghoriordan/eslint-plugin-nestjs-typed#263]
Hi,

I’m encountering an issue with the @darraghoriordan/eslint-plugin-nestjs-typed ESLint rule injectable-should-be-provided when using factory providers. Specifically, when I register an injectable service using a factory, the rule reports a false positive stating that the service is not referenced in any module’s providers array—even though it is provided.

When I define my module as follows:

**`env.module.ts` file:**

```ts
import { Global, Module } from '@nestjs/common';
import { ConfigModule, ConfigService } from '@nestjs/config';

import { envConfig } from './env.config';
import { EnvSchema } from './env.schema';
import { EnvService } from './env.service';

@Global()
@Module({
	imports: [ConfigModule.forRoot(envConfig)],
	providers: [
		{
			provide: EnvService,
			useFactory: (configService: ConfigService<EnvSchema, true>) => new EnvService(configService),
			inject: [ConfigService],
		},
	],
	exports: [EnvService],
})
export class EnvModule {}

```

**`env.service.ts` file:**

```ts
import { Injectable } from '@nestjs/common';
import { ConfigService } from '@nestjs/config';

import { EnvSchema } from './env.schema';

@Injectable()
export class EnvService {
	constructor(private configService: ConfigService<EnvSchema, true>) {}

	get<T extends keyof EnvSchema>(key: T): EnvSchema[T] {
		const value = this.configService.get(key, { infer: true });

		if (value === undefined) {
			this.throwMissingError(key);
		}

		return value;
	}

	get isProduction(): boolean {
		return this.get('NODE_ENV') === 'production';
	}

	get isDevelopment(): boolean {
		return this.get('NODE_ENV') === 'development';
	}

	get isTest(): boolean {
		return this.get('NODE_ENV') === 'test';
	}

	private throwMissingError(key: string): never {
		throw new Error(`Environment variable ${key} is missing! Check your .env file.`);
	}
}

```

as you can see, i imported `env.service.ts` to `env.module.ts`, but ESLint raises the error:

```
Expected Injectable "EnvService" to be referenced in 1 module's providers array, but found 0 references ().
If you added it already but this error still shows in your editor, please change one character in the injectable file to poke your eslint plugin.
eslint[@darraghor/nestjs-typed/injectable-should-be-provided](https://github.com/darraghoriordan/eslint-plugin-nestjs-typed/blob/main/src/docs/rules/injectable-should-be-provided.md)
```

Despite the error, EnvService is correctly provided via the custom factory provider. I’ve verified that if I switch to directly registering the service (i.e., by simply listing EnvService in the providers array), the error does not occur.

**Environment Details:**

- ESLint Plugin: @darraghoriordan/eslint-plugin-nestjs-typed:‌ 6.1.3
- NestJS version: 11.0.8
- TypeScript version: 5.7.3

Thank you for your attention.

**Comment by darraghoriordan** [comment]
Hey thanks for reporting, yea it doesn't support factory providers

- you can add a PR if you like 
- otherwise just ignore the rule for this provider
- or turn off the rule entirely
